dataprofiler/tests/profilers/test_profiler_class_options.py

```diff
@@ -0,0 +1,111 @@
+            self.assertFalse(option.structured_options.properties[key].is_enabled)        
+            option._set_helper({'structured_options.{}.is_enabled'.format(key): True}, '')
+            self.assertTrue(option.structured_options.properties[key].is_enabled)        
+
+        # Treat is_enabled as a BooleanOption
+        for key in option.structured_options.properties:
+            expected_error = "type object 'structured_options.{}.is_enabled' has no attribute 'is_enabled'".format(key)
+            with self.assertRaisesRegex(AttributeError, expected_error):
+                option._set_helper({'structured_options.{}.is_enabled.is_enabled'.format(key): True}, '')    
+        
+    def test_set(self):
+        super().test_set()
+        option = self.get_options()
+        optpth = self.get_options_path()
+
+        # Enable and Disable Options        
+        for key in option.structured_options.properties:
+            option.set({'structured_options.{}.is_enabled'.format(key): False})
+            self.assertFalse(option.structured_options.properties[key].is_enabled)        
+            option.set({'structured_options.{}.is_enabled'.format(key): True})
+            self.assertTrue(option.structured_options.properties[key].is_enabled)        
+    
+        # Treat is_enabled as a BooleanOption
+        for key in option.structured_options.properties:
+            expected_error = "type object 'structured_options.{}.is_enabled' has no attribute 'is_enabled'".format(key)
+            with self.assertRaisesRegex(AttributeError, expected_error):
+                option.set({'structured_options.{}.is_enabled.is_enabled'.format(key): True})
+    
+    def test_validate_helper(self):
+        # Valid cases should return [] while invalid cases should return a list of errors
+        option = self.get_options()
+        optpth = self.get_options_path()
+
+        # Default Configuration Is Valid
+        self.assertEqual([], option._validate_helper())
+        
+        # Variable Path Is Not A String
+        expected_error = "The variable path must be a string."
+        with self.assertRaisesRegex(ValueError, expected_error):
+            option._validate_helper(1)
+        
+        # Option is_enabled is not a boolean
+        for key in option.structured_options.properties:
+            option.set({'structured_options.{}.is_enabled'.format(key): "Hello World"}) 
+
+        expected_error = ['{}.structured_options.{}.is_enabled must be a Boolean.'.format(optpth, key) 
+            for key in option.structured_options.properties]
+        expected_error = set(expected_error)
+        self.assertSetEqual(expected_error, expected_error.intersection(set(option._validate_helper())))
+
+        # Wrong Class Type
+        option = self.get_options()
+        option.structured_options = ProfilerOptions()
+
+        expected_error = set()
+        expected_error.add('{}.structured_options must be a StructuredOptions.'.format(optpth,))
```

Comment:
This is a tad strange, would be more readable if you didn't use sets but just asserted that a singleton list was equal to the _validate_helper output, since if theres only a 1 element set, using a set at all feels unnecessary. Unless I am missing the motivation here.

Reply:
That is true. I had kept it consistent with the other assertions since they all used sets but it is very unnecessary. I will add a commit to fix this